### PR TITLE
Add ability to track users' app terms acceptance (DEV-673)

### PR DESCRIPTION
**Pull request by mikefeldberg**
DEV-673

Scope changed a bit from original ticket. We initially planned to create an endpoint for updating the term acceptance status, but it made sense to implement this as part of `update_user`. We didn't have an update user endpoint, so I added one

### CODE REVIEW

File: apps/betterangels-backend/accounts/schema.py [hunk 1]
```diff
@@ -76,6 +77,27 @@ def generate_magic_link(self, info: Info, data: MagicLinkInput) -> MagicLinkResp
...
 
+    @strawberry_django.mutation(permission_classes=[IsAuthenticated])
+    def update_current_user(self, info: Info, data: UpdateUserInput) -> UserType:
+        current_user = get_current_user(info)
+        if str(current_user.pk) != str(data.id):
+            raise PermissionError("You do not have permission to modify this user.")
+
+        user_data: dict = strawberry.asdict(data)
+
+        user = User.objects.get(id=data.id)
```
Comment (lena) [inline apps/betterangels-backend/accounts/schema.py:88]:
```suggestion
```
see above

File: apps/betterangels-backend/accounts/schema.py [hunk 2]
```diff
@@ -76,6 +77,27 @@ def generate_magic_link(self, info: Info, data: MagicLinkInput) -> MagicLinkResp
         send_magic_link(data.email, base_url)
         return MagicLinkResponse(message="Email link sent.")
 
+    @strawberry_django.mutation(permission_classes=[IsAuthenticated])
+    def update_current_user(self, info: Info, data: UpdateUserInput) -> UserType:
+        current_user = get_current_user(info)
```
Comment (lena) [inline apps/betterangels-backend/accounts/schema.py:82]: i believe this already brings back the user you're fetching on line 88, please verify
```suggestion
        user = get_current_user(info)
```

File: apps/betterangels-backend/accounts/schema.py [hunk 3]
```diff
@@ -76,6 +77,27 @@ def generate_magic_link(self, info: Info, data: MagicLinkInput) -> MagicLinkResp
...
+        if str(current_user.pk) != str(data.id):
```
Comment (lena) [inline apps/betterangels-backend/accounts/schema.py:83]:
```suggestion
        if str(user.pk) != str(data.id):
```

File: apps/betterangels-backend/accounts/tests/test_models.py [hunk 4]
```diff
@@ -23,33 +24,46 @@ def test_full_name(self) -> None:
...
 
-        remove_organization_permission_group(unauthorized_org)
+        if user_is_in_unauthorized_org:
+            unauthorized_org.add_user(user)
 
         permission_group_recipe.make(
             name="unauthorized permission group",
             organization=unauthorized_org,
         )
+        remove_organization_permission_group(unauthorized_org)
```
Comment (lena) [inline apps/betterangels-backend/accounts/tests/test_models.py:64]: a little confused how this test was working before, is this the same as not creating the permission group on line 60?

File: apps/betterangels-backend/accounts/tests/test_mutations.py [hunk 5]
```diff
@@ -65,8 +86,8 @@ def test_delete_current_user(self) -> None:
         """
 
         response = self.execute_graphql(mutation)["data"]["deleteCurrentUser"]
-        self.assertEqual(response["id"], user.pk)
-        self.assertEqual(initial_user_count, User.objects.count())
+        self.assertEqual(response["id"], self.user.pk)
+        self.assertEqual(initial_user_count - 1, User.objects.count())
```
Comment (lena) [inline apps/betterangels-backend/accounts/tests/test_mutations.py:90]: nitpick: prefer this flipped
```suggestion
        self.assertEqual(User.objects.count(), initial_user_count - 1)
```

File: apps/betterangels-backend/schema.graphql [hunk 6]
```diff
@@ -756,6 +761,8 @@ type UserType {
   username: String!
   isOutreachAuthorized: Boolean
   organizationsOrganization: [OrganizationType!]
+  hasAcceptedTos: Boolean!
```
Comment (lena) [inline apps/betterangels-backend/schema.graphql:764]: not sure if this should have the `!`

**Review by lena — COMMENTED** [review]
some comments, looks good overall

File: apps/betterangels-backend/accounts/schema.py (hunk 2 above)
Comment (mikefeldberg) [reply apps/betterangels-backend/accounts/schema.py:82]: 🤦‍♂️ peak 2am code. thanks for catching

File: apps/betterangels-backend/accounts/tests/test_models.py (hunk 4 above)
Comment (mikefeldberg) [reply apps/betterangels-backend/accounts/tests/test_models.py]: old test didn't accurately represent the property functionality. Fixed property and updated test. Thanks for drawing attention to this 🙏

File: apps/betterangels-backend/schema.graphql (hunk 6 above)
Comment (mikefeldberg) [reply apps/betterangels-backend/schema.graphql:764]: ya, it's a non-nullable field with a default

File: apps/betterangels-backend/accounts/models.py [hunk 7]
```diff
@@ -99,17 +99,22 @@ def full_name(self: "User") -> str:
...
 
-        return PermissionGroup.objects.filter(
+        user_in_authorized_org = PermissionGroup.objects.filter(
             organization__in=user_organizations, template__name__in=authorized_permission_groups
         ).exists()
 
+        if not (user_in_authorized_org and self.has_accepted_tos and self.has_accepted_privacy_policy):
+            return False
+
+        return True
```
Comment (lena) [inline apps/betterangels-backend/accounts/models.py:116]: 👀 👀 👀 👀 👀 👀 👀 👀 👀

Comment (mikefeldberg) [reply apps/betterangels-backend/accounts/models.py:116]: hahahahaha. I'm gonna walk off into the sunset now

**Review by lena — APPROVED** [review]
thanks!

**Pull request merged**